third_party/terraform/website/docs/version_3_upgrade.html.markdown

```diff
@@ -0,0 +1,161 @@
+---
+layout: "google"
+page_title: "Terraform Google Provider 3.0.0 Upgrade Guide"
+sidebar_current: "docs-google-provider-version-3-upgrade"
+description: |-
+  Terraform Google Provider 3.0.0 Upgrade Guide
+---
+
+# Terraform Google Provider 3.0.0 Upgrade Guide
+
+The `3.0.0` release of the Google provider for Terraform is a major version and
+includes some changes that you will need to consider when upgrading. This guide
+is intended to help with that process and focuses only on the changes necessary
+to upgrade from the final `2.X` series release to `3.0.0`.
+
+Most of the changes outlined in this guide have been previously marked as
+deprecated in the Terraform `plan`/`apply` output throughout previous provider
+releases, up to and including the final `2.X` series release. These changes,
+such as deprecation notices, can always be found in the CHANGELOG of the
+affected providers. [google](https://github.com/terraform-providers/terraform-provider-google/blob/master/CHANGELOG.md)
+[google-beta](https://github.com/terraform-providers/terraform-provider-google-beta/blob/master/CHANGELOG.md)
+
+## I accidentally upgraded to 3.0.0, how do I downgrade to `2.X`?
+
+If you've inadvertently upgraded to `3.0.0`, first see the
+[Provider Version Configuration Guide](#provider-version-configuration) to lock
+your provider version; if you've constrained the provider to a lower version
+such as shown in the previous version example in that guide, Terraform will pull
+in a `2.X` series release on `terraform init`.
+
+If you've only ran `terraform init` or `terraform plan`, your state will not
+have been modified and downgrading your provider is sufficient.
+
+If you've ran `terraform refresh` or `terraform apply`, Terraform may have made
+state changes in the meantime.
+
+* If you're using a *local* state, `terraform refresh` with a downgraded
```

Comment:
I think this is clearer as

"If you're using a local state, or a remote state backend that does not support versioning".

And I'd add to the end of it "if this is not sufficient, `terraform state rm` the affected resources, then `terraform import` them."